# Autogen Installation


1. Create a venv
2. Create a requirements.txt
3. install 
   #### autogen-agentchat > AssistantAgent 
   (from autogen_agentchat.agents import AssistantAgent)
   #### autogen-core >
   #### autogen-ext > autogen_ext.models.openai
   (from autogen_ext.models.openai import OpenAIChatCompletionClient)

# First Autogen Agent


In [14]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient
from dotenv import load_dotenv
import os

load_dotenv()

api_key = os.getenv('OPENAI_API_KEY')

# Connecting to the Model

In [15]:
model_client = OpenAIChatCompletionClient(model='gpt-4o',api_key=api_key)


In [16]:
agent_1 = AssistantAgent(name='my_assistant',model_client=model_client)

In [21]:
result = await agent_1.run(task='Tell me something about you?')

In [18]:
print(result)

messages=[TextMessage(id='dcc181ed-5c55-4896-9082-cde8eca24160', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 10, 17, 31, 40, 719097, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='fa08bbe9-19cc-4695-97fb-fa0cc265f7ee', source='my_assistant', models_usage=RequestUsage(prompt_tokens=42, completion_tokens=87), metadata={}, created_at=datetime.datetime(2025, 7, 10, 17, 31, 44, 117760, tzinfo=datetime.timezone.utc), content='Hello! I am an AI language model designed to assist with a wide range of queries by providing information, answering questions, and helping with problem-solving. I can handle tasks related to various subjects, including science, history, technology, and more. My goal is to assist you to the best of my abilities based on the data I have been trained on. If you have any specific questions or tasks, feel free to let me know!', type='TextMessage')] stop_reason=None


messages=[TextMessage(id='dcc181ed-5c55-4896-9082-cde8eca24160', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 10, 17, 31, 40, 719097, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='fa08bbe9-19cc-4695-97fb-fa0cc265f7ee', source='my_assistant', models_usage=RequestUsage(prompt_tokens=42, completion_tokens=87), metadata={}, created_at=datetime.datetime(2025, 7, 10, 17, 31, 44, 117760, tzinfo=datetime.timezone.utc), 
content='Hello! I am an AI language model designed to assist with a wide range of queries by providing information, answering questions, and helping with problem-solving. I can handle tasks related to various subjects, including science, history, technology, and more. My goal is to assist you to the best of my abilities based on the data I have been trained on. If you have any specific questions or tasks, feel free to let me know!', type='TextMessage')] stop_reason=None

In [22]:
result2 = await agent_1.run(task='What was my last question?')

In [23]:
result2

TaskResult(messages=[TextMessage(id='6f4d6139-0b0b-4f2c-91e3-a00fed434529', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 10, 17, 37, 1, 430010, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'), TextMessage(id='aa4899e5-b02a-4679-9d83-b2714dc66db5', source='my_assistant', models_usage=RequestUsage(prompt_tokens=307, completion_tokens=12), metadata={}, created_at=datetime.datetime(2025, 7, 10, 17, 37, 3, 37406, tzinfo=datetime.timezone.utc), content='Your last question was: "Tell me something about you?"', type='TextMessage')], stop_reason=None)

TaskResult(messages=[TextMessage(id='6f4d6139-0b0b-4f2c-91e3-a00fed434529', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 10, 17, 37, 1, 430010, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'), TextMessage(id='aa4899e5-b02a-4679-9d83-b2714dc66db5', source='my_assistant', models_usage=RequestUsage(prompt_tokens=307, completion_tokens=12), metadata={}, created_at=datetime.datetime(2025, 7, 10, 17, 37, 3, 37406, tzinfo=datetime.timezone.utc), content='Your last question was: "Tell me something about you?"', type='TextMessage')], stop_reason=None)